benchmarks/decoders/benchmark_decoders_library.py

```diff
@@ -459,55 +479,57 @@ def run_benchmarks(
     for decoder_name, decoder in decoder_dict.items():
         for video_file_path in video_files_paths:
             print(f"video={video_file_path}, decoder={decoder_name}")
-            # We only use the VideoDecoder to get the metadata and get
-            # the list of PTS values to seek to.
-            simple_decoder = VideoDecoder(video_file_path)
-            duration = simple_decoder.metadata.duration_seconds
-            pts_list = [
-                i * duration / num_uniform_samples for i in range(num_uniform_samples)
+            metadata = get_metadata(video_file_path)
+            metadata_label = f"{metadata.codec} {metadata.width}x{metadata.height}, {metadata.duration_seconds}s {metadata.average_fps}fps"
+
+            duration = metadata.duration_seconds
+            uniform_pts_list = [
+                i * duration / num_samples for i in range(num_samples)
             ]
-            metadata = simple_decoder.metadata
-            metadata_string = f"{metadata.codec} {metadata.width}x{metadata.height}, {metadata.duration_seconds}s {metadata.average_fps}fps"
-            if verbose:
-                print(
-                    f"video={video_file_path}, decoder={decoder_name}, pts_list={pts_list}"
+
+            random_pts_list = (torch.rand(num_samples) * duration).tolist()
```

Comment:
What's the random seed here? Is this the source of the benchmark run time variation?

Reply:
Oh, good point. We're doing a _different_ random list for all decoders. We should fix that. I can also set the seed to be deterministic. Maybe in the future that should be a benchmark parameter.